sdk/core/System.ClientModel/src/ModelReaderWriter/ModelReaderWriterOptions.cs

```diff
@@ -24,13 +28,118 @@ public class ModelReaderWriterOptions
     /// Initializes a new instance of <see cref="ModelReaderWriterOptions"/>.
     /// </summary>
     /// <param name="format">The format to read and write models.  Pass in 'W' to use the service defined wire format.</param>
-    public ModelReaderWriterOptions (string format)
+    public ModelReaderWriterOptions(string format)
     {
         Format = format;
     }
 
+    internal ModelReaderWriterOptions(ModelReaderWriterOptions options)
+    {
+        Format = options.Format;
+        _proxies = options._proxies;
+        IsCoreOwned = true;
+    }
+
+    internal bool HasProxies => _proxies is not null;
```

Comment:
should we also check that there are any Keys in the dictionary?

Reply:
We only construct it when Add is called so checking null should be enough.  I could change to this if you prefer?

```c#
internal bool HasProxies => _proxies?.Count > 0;
```

Reply:
Just in case we ever initialize differently in the future, it may be nice to validate.